src/main/java/org/littleshoot/proxy/extras/SelfSignedSslEngineSource.java

```diff
@@ -150,8 +234,7 @@ public void checkServerTrusted(X509Certificate[] arg0,
             sslContext = SSLContext.getInstance(PROTOCOL);
             sslContext.init(keyManagers, trustManagers, null);
         } catch (final Exception e) {
-            throw new Error(
-                    "Failed to initialize the server-side SSLContext", e);
+            throw new Error("Failed to initialize the server-side SSLContext", e);
```

Comment:
It's definitely a serious problem and the client *probably* can't do anything about it. (Although we shouldn't assume -- what if the problem was an IOException because the user didn't have write permissions on the keystore file, or it was on a shared drive that is not accessible? Maybe the client wants to try another keystore file. It's not our place to say.) But that's what an unchecked exception is for. If you look at the Error subclasses -- VirtualMachineError, LinkageError, etc., it's clear they're not in the same league as this.

I think it's possible to make a case that there are just too many checked exceptions in this code to catch individually, so you catch Exception as a convenience, but even then you'd want to wrap it into a RuntimeException subclass with an appropriate message, instead of an Error.

The same comment applies to the new clientSslEngineFor() method above, and given the similarity of the exception types between these methods, I think it makes sense to be consistent. You didn't create the Error statement, but since you're modifying this class extensively, including its catch block, making this consistent could be considered 'being a good neighbor'.